### PR TITLE
c-ares: add version 1.19.0

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **c-ares/***

- add version 1.19.0
- add patch description
- use `add_subdirectory` in test_v1_package
- format conandata.yml

Try to close https://github.com/conan-io/conan-center-index/issues/15564.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by jwillikers — DISMISSED** [review]

**Review by mlappo — APPROVED** [review]
LGTM

File: recipes/c-ares/all/conandata.yml [hunk 1]
```diff
@@ -1,29 +1,47 @@
...
   "1.15.0":
-    sha256: 7deb7872cbd876c29036d5f37e30c4cbc3cc068d59d8b749ef85bb0736649f04
-    url: https://github.com/c-ares/c-ares/archive/cares-1_15_0.tar.gz
+    url: "https://github.com/c-ares/c-ares/archive/cares-1_15_0.tar.gz"
+    sha256: "7deb7872cbd876c29036d5f37e30c4cbc3cc068d59d8b749ef85bb0736649f04"
   "1.14.0":
-    sha256: 62dd12f0557918f89ad6f5b759f0bf4727174ae9979499f5452c02be38d9d3e8
-    url: https://github.com/c-ares/c-ares/archive/cares-1_14_0.tar.gz
+    url: "https://github.com/c-ares/c-ares/archive/cares-1_14_0.tar.gz"
+    sha256: "62dd12f0557918f89ad6f5b759f0bf4727174ae9979499f5452c02be38d9d3e8"
```
Comment (SpaceIm) [inline recipes/c-ares/all/conandata.yml:22]: Would it be possible to take advantage of this modification to point to official tarballs (as already done for 1.18.1 & 1.19.0)?

Comment (toge) [reply recipes/c-ares/all/conandata.yml]: @SpaceIm 
Thanks a lot!
I modified urls.

**Review by jwillikers — APPROVED** [review]

File: recipes/c-ares/all/conandata.yml [hunk 2]
```diff
@@ -1,29 +1,47 @@
...
   "1.17.1":
-    sha256: 61f7cf09605f5e38d4828f82d0e2ddb9de8e355ecfd6819b740691c644583b8f
-    url: https://github.com/c-ares/c-ares/archive/cares-1_17_1.tar.gz
+    url: "https://github.com/c-ares/c-ares/releases/download/cares-1_17_1/c-ares-1.17.1.tar.gz"
+    sha256: "d73dd0f6de824afd407ce10750ea081af47eba52b8a6cb307d220131ad93fc40"
   "1.16.1":
-    sha256: 870962cc8f6b352303c404ce848e2ea1f1072f3c0a940042209a72179511c08c
-    url: https://github.com/c-ares/c-ares/archive/cares-1_16_1.tar.gz
+    url: "https://github.com/c-ares/c-ares/releases/download/cares-1_16_0/c-ares-1.16.0.tar.gz"
+    sha256: "de058ad7c128156e2db6dc98b8a359924d6f210a1b99dd36ba15c8f839a83a89"
```
Comment (SpaceIm) [inline recipes/c-ares/all/conandata.yml:16]: should be https://github.com/c-ares/c-ares/releases/download/cares-1_16_1/c-ares-1.16.1.tar.gz ;)

File: recipes/c-ares/all/conandata.yml (hunk 1 above)
Comment (SpaceIm) [reply recipes/c-ares/all/conandata.yml]: thanks, just one url to fix: https://github.com/conan-io/conan-center-index/pull/15523#discussion_r1092415978

**Review by jwillikers — APPROVED** [review]

File: recipes/c-ares/all/conandata.yml (hunk 2 above)
Comment (toge) [reply recipes/c-ares/all/conandata.yml]: @SpaceIm 
Sorry for my miss.
Fixed.

**Review by SpaceIm — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Review by miklelappo — APPROVED** [review]
LGTM

**Pull request merged**